pkg/ip/addr_linux.go

```diff
@@ -57,12 +63,17 @@ func SettleAddresses(ifName string, timeout int) error {
 		if ok {
 			return nil
 		}
-		if time.Now().After(deadline) {
+
+		go func() {
+			timeEventChan <- time.After(SETTLE_INTERVAL)
```

Comment:
time.After returns a channel... immediately. This isn't actually waiting.

Reply:
sorry, too late to reply, I still try to figure out the root cause, and I sometimes encounter this problem after applying this fix on a new k8s environment.

+++ time.After returns a channel... immediately. This isn't actually waiting.  
yes, it will be blocked by the select sentence